andela/learning_map_api
# #151145080 Set up server-side event stream

#### What does this PR do?
- Add Server sent events to the application.

#### Description of Task to be completed?
- Add  ```/api/v1/notification/stream``` url for stream event handshake & subscription
- Change startup command in Procfile

#### How should this be manually tested?
- Install [redis](https://medium.com/@petehouston/install-and-config-redis-on-mac-os-x-via-homebrew-eb8df9a4f298)
- Start redis 
- In your .env file add ```REDIS_URL=redis://localhost```
- pip install requirement.txt

From root folder runserver
```
$ gunicorn main:app  --worker-class gevent
```
From browser open dev console and add this:
```
var source = new EventSource("http://127.0.0.1:8000/api/v1/notification/stream");

source.addEventListener('testing', function(notification){ console.log(JSON.parse(notification.data))});
```
 Send a request to any invalid url e.g http://127.0.0.1:8000/notfound

A message should appear on your dev console.

#### What are the relevant pivotal tracker stories?
- #151145080

#### Screenshots (if appropriate)
<img width="1009" alt="screen shot 2017-09-21 at 20 50 09" src="https://user-images.githubusercontent.com/17669596/30710855-8a3cdefe-9f0f-11e7-8110-1bbdf7ff7891.png">
